odoo/tools/sql.py

```diff
@@ -119,22 +120,40 @@ def rename_column(cr, tablename, columnname1, columnname2):
     cr.execute('ALTER TABLE "{}" RENAME COLUMN "{}" TO "{}"'.format(tablename, columnname1, columnname2))
     _schema.debug("Table %r: renamed column %r to %r", tablename, columnname1, columnname2)
 
-def convert_column(cr, tablename, columnname, columntype):
+def convert_column(cr, tablename, columnname, columntype, fromtype=None):
     """ Convert the column to the given type. """
+    # TODO VSC: I don't understand why we drop the index.
+    drop_index(cr, tablename+'_'+columnname+'_index', tablename)
+
+    if (fromtype != 'jsonb') and (columntype != 'jsonb'):
+        try:
+            with cr.savepoint(flush=False):
+                cr.execute('ALTER TABLE "{}" ALTER COLUMN "{}" TYPE {}'.format(tablename, columnname, columntype),
+                           log_exceptions=False)
+            _schema.debug("Table %r: column %r altered to type %s", tablename, columnname, columntype)
+            return
+        except (psycopg2.NotSupportedError, psycopg2.errors.DatatypeMismatch):
+            # simple convert failed so we have to do it the hard way
+            pass
+
+    temp_name = columnname + '_old'
+    while column_exists(cr, tablename, temp_name):
+        temp_name += '_'
+    rename_column(cr, tablename, columnname, temp_name)
+    create_column(cr, tablename, columnname, columntype)
+    if fromtype == 'jsonb':
+        query = 'UPDATE "{0}" SET "{1}"="{3}"->>\'en_US\'::{2};'
+    elif columntype == 'jsonb':
+        query = 'UPDATE "{0}" SET "{1}"=json_build_object(\'en_US\', {3}::varchar) WHERE {3} IS NOT NULL;'
```

Comment:
This code assumes that the column type `jsonb` means "translated field".
But we now have two new field types that use column type `jsonb` https://github.com/odoo/odoo/pull/95184.
It is therefore incorrect.
This crap must be moved to the field method, or we should make a specific `convert_column` for translated fields...